lib/flipper/api/middleware.rb

```diff
@@ -9,10 +9,16 @@
 module Flipper
   module Api
     class Middleware
-      def initialize(app, options = {})
+      def initialize(app, flipper = nil, options = {})
         @app = app
         @env_key = options.fetch(:env_key, 'flipper')
 
+        if flipper.respond_to?(:call)
+          @flipper_block = flipper
+        else
+          @flipper = flipper || Flipper
+        end
```

Comment:
Per discussion in #526, I would be in favor of either passing this in `options`, or removing it entirely and relying on `SetupEnv` in all the middleware/rack apps.